tools/dds/dds-sniffer/dds-sniffer.hpp

```diff
@@ -72,4 +85,21 @@ class Sniffer : public eprosima::fastdds::dds::DomainParticipantListener
                                        const eprosima::fastrtps::string_255 topic_name,
                                        const eprosima::fastrtps::string_255 type_name,
                                        const eprosima::fastrtps::types::TypeInformation & type_information ) override;
+
+    // Topics data-base functions
+    void save_topic_writer( const eprosima::fastrtps::rtps::WriterDiscoveryInfo & info );
+    void remove_topic_writer( const eprosima::fastrtps::rtps::WriterDiscoveryInfo & info );
+    void save_topic_reader( const eprosima::fastrtps::rtps::ReaderDiscoveryInfo & info );
+    void save_max_indentation( const std::string && str );
+    void remove_topic_reader( const eprosima::fastrtps::rtps::ReaderDiscoveryInfo & info );
+
+    // Helper print functions
```

Comment:
This is fine for now, but in general I want this class to be usable from unit-testing (LibCI) -- so eventually I want to leave the printing to the code using this class.

Reply:
Will be refactored in a future commit